core/filter.js

```diff
@@ -59,19 +59,29 @@
 	 * {@link CKEDITOR.filter.allowedContentRules} instead of {@link CKEDITOR.editor}
 	 * to the constructor:
 	 *
-	 *		var filter = new CKEDITOR.filter( 'b' );
+	 * ```javascript
+	 * var filter = new CKEDITOR.filter( 'b' );
 	 *
-	 *		filter.check( 'b' ); // -> true
-	 *		filter.check( 'i' ); // -> false
-	 *		filter.allow( 'i' );
-	 *		filter.check( 'i' ); // -> true
+	 * filter.check( 'b' ); // -> true
+	 * filter.check( 'i' ); // -> false
+	 * filter.allow( 'i' );
+	 * filter.check( 'i' ); // -> true
+	 * ```
+	 *
+	 * If filter is only used by a single editor instance, you should pass additional owner editor instance
+	 * so the filter could be removed with {@link CKEDITOR.editor#destroy} method.
+	 *
+	 * ```javascript
+	 * var filter = new CKEDITOR.filter( 'b', editor );
+	 * ```
 	 *
 	 * @since 4.1
 	 * @class
 	 * @constructor Creates a filter class instance.
 	 * @param {CKEDITOR.editor/CKEDITOR.filter.allowedContentRules} editorOrRules
+	 * @param {CKEDITOR.editor} owner Editor owning the filter instance.
```

Comment:
I agree that it's not beautiful. But let's say you would like to create empty filter and setup it by `CKEDITOR.filter#allow` function e.g
```javascript
var filter = new CKEDITOR.filter();
filter.allow( {...} );
```
In that case, you won't be able to create a filter with owning `editor` - if you try to pass `editor` as an argument you will create a filter based on `editor` (https://github.com/ckeditor/ckeditor-dev/blob/major/core/filter.js#L168).

We have that case in our code (https://github.com/ckeditor/ckeditor-dev/blob/major/plugins/clipboard/plugin.js#L1347) and it's possible that people could use it in that way also.

Reply:
I agree it can be done but with that change, we will provide breaking change for our API. If we are ok with that I also prefer `new CKEDITOR.filter( editorOrRules, rules )` signature. WDYT @Comandeer?